src/icp_canister_backend/tests/rewards_tests.rs

```diff
@@ -90,73 +86,310 @@ fn test_reward_rate_increase_decrease_during_episodes() {
+    let second_reward_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    reward_pool(&pic, canister_id, ledger_id, user, second_reward.clone())
+        .expect("Second reward pool should succeed");
+
+    advance_time(&pic, EPISODE_DURATION / 4);
+    let third_reward_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    reward_pool(&pic, canister_id, ledger_id, user, third_reward.clone())
+        .expect("Third reward pool should succeed");
+    
+    let first_end = ((first_reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION + 1) * EPISODE_DURATION;
+    let second_end = ((second_reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION + 1) * EPISODE_DURATION;
+    let third_end = ((third_reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION + 1) * EPISODE_DURATION;
+    
+    let latest_end = first_end.max(second_end).max(third_end);
+    let current_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    
+    if latest_end > current_time {
+        advance_time(&pic, latest_end - current_time);
+    }
+    
+    let final_rewards = pic
+        .query_call(canister_id, user, "get_deposits_rewards", encode_args((vec![0u64],)).unwrap())
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get final rewards");
+    
+    assert_with_error!(&final_rewards, &total_expected_rewards, &ALLOWED_ERROR, "Multiple reward additions final total");
 }
+#[test]
+fn test_reward_distribution_between_large_and_small_deposits() {
+    let (pic, canister_id, ledger_id) = setup();
+    let large_depositor = Principal::from_text("rdmx6-jaaaa-aaaaa-aaadq-cai").unwrap();
+    let small_depositor = Principal::from_text("rrkah-fqaaa-aaaaa-aaaaq-cai").unwrap();
+
+    let large_deposit = Nat::from(1_000_000_000u64); // 10 BTC
+    let small_deposit = icp_canister_backend::MINIMUM_DEPOSIT_AMOUNT.clone() + Nat::from(1u64);
+    let reward_amount = Nat::from(1_000u64); // $1
+
+    let episode = get_stakable_episode(&pic, canister_id, 7);
+
+    create_deposit(&pic, canister_id, ledger_id, large_depositor, large_deposit, episode);
+    create_deposit(&pic, canister_id, ledger_id, small_depositor, small_deposit, episode);
+
+    let reward_start_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    reward_pool(&pic, canister_id, ledger_id, large_depositor, reward_amount.clone())
+        .expect("Reward pool should succeed");
+
+    let last_reward_episode = (reward_start_time + EPISODE_DURATION * 12) / EPISODE_DURATION;
+    let exact_reward_duration = (last_reward_episode + 1) * EPISODE_DURATION - reward_start_time;
+    advance_time(&pic, exact_reward_duration);
+
+    let large_depositor_rewards = pic
+        .query_call(canister_id, large_depositor, "get_deposits_rewards", encode_args((vec![0u64],)).unwrap())
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get large depositor rewards");
+    
+    let small_depositor_rewards = pic
```

Comment:
Check out what is actual small_depositor_rewards value, I believe in the current configuration, it may be 0. Make sure that the expected value is at least 10 sats.
Also, for such small values, ALLOWED_ERROR must be adjusted, for example, if the expected reward is 10 sats, then if we have ALLOWED_ERROR equal 10, this may lead to missing the situation when a user had to get 10 sats but received 0.
To mitigate this situation, put in assert_with_error! even lower allowed error value like 1 sat .